packages/cli/src/commands/upgrade.js

```diff
@@ -131,6 +132,25 @@ const runUpgrade = ({ d, tag }) => {
   ]
 }
 
+const SEMVER_REGEX = /(?<=^v?|\sv?)(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-(?:0|[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*)(?:\.(?:0|[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*))*)?(?:\+[\da-z-]+(?:\.[\da-z-]+)*)?(?=$|\s)/gi
+const validateTag = (tag) => {
+  const isTagValid = tag === 'rc' || tag === 'canary' || tag === 'latest' || SEMVER_REGEX.test(tag)
+
+  if (!isTagValid) {
+    // Output is quite long, this highlights the error a little bit
+    console.log(
+      c.error(
+        'Invalid tag supplied. Supported values: rc, canary or valid semver version'
+      )
+    )
+
+    // Stop execution
+    throw new Error('Invalid tag')
+  }
```

Comment:
```suggestion
  if (!isTagValid) {
    // Stop execution
    throw new Error(
      c.error(
        'Invalid tag supplied. Supported values: rc, canary, latest, or valid semver version\n'
      )
    )
  }
```

remove duplicate error